src/main/preload/externalAPI.ts

```diff
@@ -0,0 +1,459 @@
+ */
+
+// Disabling no-explicit-any for this legacy code
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+window.addEventListener('message', ({origin, data = {}}: {origin?: string; data?: {type?: string; message?: any}} = {}) => {
+    const {type, message = {}} = data;
+    if (origin !== window.location.origin) {
+        return;
+    }
+    switch (type) {
+    case 'webapp-ready':
+    case 'get-app-version': {
+        // register with the webapp to enable custom integration functionality
+        ipcRenderer.invoke(GET_APP_INFO).then((info) => {
+            console.log(`registering ${info.name} v${info.version} with the server`);
+            window.postMessage(
+                {
+                    type: 'register-desktop',
+                    message: info,
+                },
+                window.location.origin || '*',
+            );
+        });
+        break;
+    }
+    case 'dispatch-notification': {
+        const {title, body, channel, teamId, url, silent, data: messageData} = message;
+        channels.set(channel.id, channel);
+        ipcRenderer.send(NOTIFY_MENTION, title, body, channel.id, teamId, url, silent, messageData.soundName);
+        break;
+    }
+    case BROWSER_HISTORY_PUSH: {
+        const {path} = message as {path: string};
+        ipcRenderer.send(BROWSER_HISTORY_PUSH, path);
+        break;
+    }
+    case 'history-button': {
+        ipcRenderer.invoke(REQUEST_BROWSER_HISTORY_STATUS).then(sendHistoryButtonReturn);
+        break;
+    }
+    case CALLS_LINK_CLICK: {
+        ipcRenderer.send(CALLS_LINK_CLICK, message.link);
+        break;
+    }
+    case GET_DESKTOP_SOURCES: {
+        ipcRenderer.invoke(GET_DESKTOP_SOURCES, message).then(sendDesktopSourcesResult);
+        break;
+    }
+    case CALLS_WIDGET_SHARE_SCREEN: {
+        ipcRenderer.send(CALLS_WIDGET_SHARE_SCREEN, message.sourceID, message.withAudio);
+        break;
+    }
+    case CALLS_JOIN_CALL: {
+        ipcRenderer.invoke(CALLS_JOIN_CALL, message).then(sendCallsJoinedCall);
+        break;
+    }
+    case CALLS_JOINED_CALL: {
+        ipcRenderer.send(CALLS_JOINED_CALL, message.callID);
+        break;
+    }
+    case CALLS_JOIN_REQUEST: {
+        ipcRenderer.send(CALLS_JOIN_REQUEST, message.callID);
+        break;
+    }
+    case CALLS_WIDGET_RESIZE: {
+        ipcRenderer.send(CALLS_WIDGET_RESIZE, message.width, message.height);
+        break;
+    }
+    case CALLS_ERROR: {
+        ipcRenderer.send(CALLS_ERROR, message.error, message.callID, message.errMsg);
+        break;
+    }
+    case CALLS_WIDGET_CHANNEL_LINK_CLICK:
+    case CALLS_LEAVE_CALL:
+    case DESKTOP_SOURCES_MODAL_REQUEST:
+    case CALLS_POPOUT_FOCUS: {
+        ipcRenderer.send(type);
+    }
+    }
+});
+
+// Legacy support to hold the full channel object so that it can be used for the click event
+const channels: Map<string, {id: string}> = new Map();
```

Comment:
Unfortunately even though the channelId isn't needed, older versions of the Web and Desktop Apps both expect the full `Channel` object to be passed through.

Rather than just accept that going forward we'll always do that, I've added this temporary fix to store and send back the full object for older versions.